packages/redis/src/index.ts

```diff
@@ -152,6 +157,20 @@ export default (req: Request, res: Response, next: NextFunction) => {
         await client.set(key, JSON.stringify(obj));
       }
     }
+    if (zencode.match(Action.DEL)) {
+      client = client || (await getRedisClient());
+      const chkParams = zencode.chunkedParamsOf(Action.DEL, 1);
+      for(const [keyName] of chkParams) {
+        const key = result[keyName] || content[keyName];
+        if(!key) {
+          throw new Error(`${keyName} not defined`)
+        }
+      }
+      for(const [keyName] of chkParams) {
+        const key = result[keyName] || content[keyName];
+        await client.del(key);
+      }
```

Comment:
Why the double for loop?
Can't be:
```
        const key = result[keyName] || content[keyName];
        if(!key) {
          throw new Error(`${keyName} not defined`)
        }
        await client.del(key);
```

Reply:
Well, maybe, if there is an error nothing has to be deleted. So I look at all the keys and if there are all good, then I start deleting